databroker/core.py

```diff
@@ -0,0 +1,543 @@
+from __future__ import print_function
+import six  # noqa
+from collections import defaultdict
+from itertools import chain
+import pandas as pd
+import tzlocal
+import doct as doc
+from pims import FramesSequence, Frame
+from metadatastore.commands import NoRunStop, NoEventDescriptors
+from filestore import handlers, HandlerBase
+import logging
+import numbers
+
+# Toolz and CyToolz have identical APIs -- same test suite, docstrings.
+try:
+    from cytoolz.dicttoolz import merge
+except ImportError:
+    from toolz.dicttoolz import merge
+
+
+logger = logging.getLogger(__name__)
+
+
+def fill_event(fs, event, handler_registry=None, handler_overrides=None):
+    """
+    Populate events with externally stored data.
+
+    Parameters
+    ----------
+    fs : FileStoreRO
+    event : document
+    handler_registry : dict, optional
+        mapping spec names (strings) to handlers (callable classes)
+    handler_overrides : dict, optional
+        mapping data keys (strings) to handlers (callable classes)
+    """
+    if handler_overrides is None:
+        handler_overrides = {}
+    is_external = _inspect_descriptor(event.descriptor)
+    mock_registries = {data_key: defaultdict(lambda: handler)
+                       for data_key, handler in handler_overrides.items()}
+    for data_key, value in six.iteritems(event.data):
+        if is_external.get(data_key, False):
+            if data_key not in handler_overrides:
+                with fs.handler_context(handler_registry) as _fs:
+                    event.data[data_key] = _fs.get_datum(value)
+            else:
+                mock_registry = mock_registries[data_key]
+                with fs.handler_context(mock_registry) as _fs:
+                    event.data[data_key] = _fs.get_datum(value)
+
+
+class Header(doc.Document):
+    """A dictionary-like object summarizing metadata for a run."""
+
+    @classmethod
+    def from_run_start(cls, mds, run_start, verify_integrity=False):
+        """
+        Build a Header from a RunStart Document.
+
+        Parameters
+        ----------
+        run_start : metadatastore.document.Document or str
+            RunStart Document
+
+        Returns
+        -------
+        header : databroker.broker.Header
+        """
+        if isinstance(run_start, six.string_types):
+            run_start = mds.run_start_given_uid(run_start)
+        run_start_uid = run_start['uid']
+
+        try:
+            run_stop = doc.ref_doc_to_uid(mds.stop_by_start(run_start_uid),
+                                          'run_start')
```

Comment:
I think it should be 'run_start'. This line is dereferencing the 'run_start' field of the 'run_stop' document.